packages/babel-generator/src/printer.js

```diff
@@ -165,11 +167,26 @@ export default class Printer {
   }
 
   /**
+   * Inserts a new line when format.maxLineLength is set
+   */
+
+  _ensureLineLength() {
+    if (this.format.maxLineLength && this._buf.getCurrentColumn() >= this.format.maxLineLength) {
+      this._newline();
+    }
+  }
+
+  /**
    * Add a newline (or many newlines), maintaining formatting.
    */
 
   newline(i?: number): void {
-    if (this.format.retainLines || this.format.compact) return;
+    if (this.format.retainLines) return;
+
+    if (this.format.compact) {
```

Comment:
My only concern would be affecting performance in non-compact mode. If you don't see any change, I see no problem with enabling this for non-compact too.